src/init.cpp

```diff
@@ -1048,6 +1048,9 @@ bool AppInitParameterInteraction()
     fAcceptDatacarrier = GetBoolArg("-datacarrier", DEFAULT_ACCEPT_DATACARRIER);
     nMaxDatacarrierBytes = GetArg("-datacarriersize", nMaxDatacarrierBytes);
 
+    nLocalServices = ServiceFlags(nLocalServices | NODE_SEGWIT2X);
```

Comment:
Right, for preferential peering hard-setting the bit is definitely the simplest solution.